Release Day III

joegreen2 opened the pull request:
Release Day III

buchholz-sean committed:
Create usage documentation

Documentation for Max Bot usage once installed. Includes a link in README.md to the usage doc.

buchholz-sean committed:
Update README.md

Added information about Nodemailer module. Also replaced a reference to `.env` file with `max.config.js` which is current correct env config file.

buchholz-sean committed:
Merge pull request #88 from reactivepixel/feature-verify-documentation

Create usage documentation

Rhuansantos committed:
recording user id instead of username

the username can be change but user id will stay the same

Commit:
Added instructions to email

Commit:
instructions fix

Rhuansantos committed:
creating function to find the crew role and add the user if the verify process it's ok

buchholz-sean committed:
Create custom exponent method

Custom method now generates verification codes of correct specified length (replaces hard-coded max value)

Rhuansantos committed:
Merge pull request #89 from reactivepixel/user-roles

User roles

buchholz-sean committed:
Merge pull request #90 from reactivepixel/feature-code-gen-length

Create custom exponent method

Commit:
Merge pull request #91 from reactivepixel/email-template

Email template

MROwens commented on usage.md:
```diff
@@ -0,0 +1,72 @@
+
+Max also includes a couple of special commands just for Officers and above for making announcements to multiple channels.
+
+### !shout
```
Should we remove the !shout and !shoutAll since the functionality will not be in this release?

buchholz-sean replied:
Agreed. Let's drop this.

buchholz-sean commented on bot/controllers/verify.js:
```diff
@@ -53,11 +62,20 @@ module.exports = () => {
+                  if (guild[el].name === 'crew') {
+                    m.member.addRole(guild[el].id);
+                  } else {
+                    util.log('the moredator has to create the crew role');
```
Typo: moredator -> moderator; Also, no `output` and `level` params passed to `util.log()`. Recommend `util.log(<message>, el, 2);`

buchholz-sean commented on bot/controllers/verify.js:
```diff
@@ -53,11 +62,20 @@ module.exports = () => {
                   verified: 1,
-                }).then(util.log).catch(util.error);
+                });
+                // maping guild roles to find the crew role id
```
Typo: maping -> mapping

buchholz-sean commented on bot/controllers/verify.js:
```diff
@@ -41,6 +49,7 @@ module.exports = () => {
         if (validDomains.includes(emailDomain)) {
           const codeLength = 6;
           const code = generateCode(codeLength);
+          util.log('code', code);
```
No debug level param in `util.log()`.

buchholz-sean reviewed (changes requested):
Some minor typos. A couple issues with missing params in `util.log()`s. 

Need to change Line 57 value to `600000`

Commit:
Usage.md shout fix - role change error fix

Commit:
Util.log level added - mapping comment fix

Commit:
Merge branch 'release' into dev

buchholz-sean committed:
Change amount of time for collector

Set collector time to 600000ms (10min) instead of 15000ms (15s) used
in testing.
